apps/leo_storage/src/leo_storage_handler_del_directory.erl

```diff
@@ -396,22 +400,20 @@ check_stats_1([_|DelBucketStateList], From, State) ->
 check_stats_2(#del_dir_state{mq_id = MQId,
                              type = Type,
                              directory = Directory,
-                             state = ?STATE_PENDING}, From,_State) when MQId /= null,
-                                                                        Type == ?TYPE_DEL_DIR ->
+                             state = ?STATE_PENDING,
+                             enqueued_at = EnqueuedAt} = CurState, From,_State) when MQId /= null,
+                                                                                     Type == ?TYPE_DEL_DIR ->
     case leo_mq_api:count(MQId) of
         {ok, 0} ->
             _Pid = spawn(
                      fun() ->
-                             insert_messages(From, MQId, Type, Directory)
+                             insert_messages(From, MQId, Type, Directory, EnqueuedAt)
                      end),
             ok;
         {ok,_Count} ->
-            update_state(#del_dir_state{
-                            mq_id = MQId,
-                            directory = Directory,
-                            type = Type,
-                            state = ?STATE_MONITORING,
-                            timestamp = leo_date:now()});
+            update_state(CurState#del_dir_state{
```

Comment:
mq_id, directory, type also should be updated here as it's fully overwrite operation on KVS so fields without assigning any value are set to the default on that erlang record.